chain-core/src/init/coin.rs

```diff
@@ -50,11 +50,16 @@ impl Coin {
         Coin(0)
     }
 
-    /// create of unitary coin (a coin of value `1`)
+    /// create of base unitary coin (a coin of value `1`)
     pub fn unit() -> Self {
         Coin(1)
     }
 
+    /// create of non-base coin of value 1 (assuming 8 decimals)
+    pub fn one() -> Self {
+        Coin(100_000_000)
```

Comment:
Since `MAX_COIN` is configurable, it is possible that it is smaller than `100_000_000`. Should we use `Coin::new(100_000_000).expect("error")` instead? Also, you can mark these functions as `const`.

Reply:
MAX_COIN isn't configurable -- it's a fixed constant (the maximum supply set in the ERC20 contract)

Reply:
or could the maximum supply ever change @CeruleanAtMonaco ?

Reply:
moved the number to `const`

Reply:
No plan to change maximum supply.